field/src/field.rs

```diff
@@ -265,6 +265,35 @@ pub trait FieldAlgebra:
     }
 }
 
+/// A ring implements `InjectiveMonomial<N>` if the algebraic function
+/// `f(x) = x^N` is an injective map on elements of the ring.
+///
+/// We do not enforce that this map be invertible as there are useful
+/// cases such as polynomials or symbolic expressions where no inverse exists.
```

Comment:
Hmm, if a `FieldAlgebra` were finite, it seems like we could just say that an injective endofunction over a finite set must be a permutation?

The `SymbolicExpression` case is a bit special since it's not necessarily finite; it seems there are various ways we could define it...
1. The infinite, non-associative algebra of binary expression trees, where e.g. `(x + x) + x`, `x + (x + x)` and `3 * x` are all distinct elements.
2. Some modified version of (1) to make it associative.
3. Some kind of multivariate polynomial-ring-like algebra, maybe one which treats polynomials as identical if they have the same behavior given base field inputs.